### ISSUE TITLE
Check for importing named exports from CJS module

### ISSUE THREAD

**Issue by graingert** [benmosher/eslint-plugin-import#928]
With the new node.js 8.5 ESM support importing named CJS modules is no longer supported.

Eg:

```js
// bad
import { get } from 'lodash';

//good
import _ from 'lodash';
const get = _.get;

//good
import { get } from 'lodash-es';
```

**Comment by ljharb** [comment]
This should be the case *already* - the fact that Babel allows it is a quirk of their implementation, and not something this plugin should allow.

**Comment by graingert** [comment]
This plugin should prevent it

**Comment by ljharb** [comment]
Right, sorry, what I mean to say is that this should have always been the default behavior, and an *option* should have been required to allow named imports from CJS packages (probably a separate one for node_modules and for project code).

What we should do now is add these options, but inverted - so that you can *enable* the correct and stricter checking.

**Comment by graingert** [comment]
To clarify (as on IRC), this is technically already case in `named`.

> A module path that is ignored or not unambiguously an ES module will not be reported when imported.

This should be tweaked so that importing named imports from CJS files errors

**Comment by ljharb** [comment]
Indeed - this should be done via an option so as not to be a breaking change. Additionally, this option should be able to differentiate between project code, and node_modules code, since you can't control the latter's use of syntax.

**Comment by klarkc** [comment]
Just to add something new here:

According official [node documentation](https://nodejs.org/api/esm.html#esm_mandatory_file_extensions), in ESM, file extensions are mandatory, with default value for this rule, it breaks Node native behavior.

**Comment by ljharb** [comment]
@klarkc file extensions *for relative paths* are mandatory. For packages, "exports" can (and always should) provide an extensionless specifier.
